internal/worker/cache.go

```diff
@@ -178,21 +178,14 @@ func (c *cache) HandleEvent(event webhooks.Event) (err error) {
 	case api.EventConsensusUpdate:
 		log = log.With("bh", e.BlockHeight, "ts", e.Timestamp)
 		c.handleConsensusUpdate(e)
+	case api.EventSettingUpdate:
+		log = log.With("gouging", e.GougingSettings != nil, "pinned", e.PinnedSettings != nil, "upload", e.UploadSettings != nil, "ts", e.Timestamp)
+		c.handleSettingUpdate(e)
 	case api.EventContractAdd:
-		log = log.With("fcid", e.Added.ID, "ts", e.Timestamp)
-		c.handleContractAdd(e)
 	case api.EventContractArchive:
-		log = log.With("fcid", e.ContractID, "ts", e.Timestamp)
-		c.handleContractArchive(e)
 	case api.EventContractRenew:
-		log = log.With("fcid", e.Renewal.ID, "renewedFrom", e.Renewal.RenewedFrom, "ts", e.Timestamp)
-		c.handleContractRenew(e)
 	case api.EventHostUpdate:
-		log = log.With("hk", e.HostKey, "ts", e.Timestamp)
-		c.handleHostUpdate(e)
-	case api.EventSettingUpdate:
-		log = log.With("gouging", e.GougingSettings != nil, "pinned", e.PinnedSettings != nil, "upload", e.UploadSettings != nil, "ts", e.Timestamp)
-		c.handleSettingUpdate(e)
+		c.cache.Invalidate(cacheKeyUsableHosts)
```

Comment:
Took a bit of a shortcut here because we're going to drastically overhaul the cache anway.